### ISSUE TITLE
Fix compaction logic to tolerate gaps in the sequence of Level 0 delta files

### ISSUE THREAD

**Issue by hlinnaka** [neondatabase/neon#1730]
We observed a project on staging, which had delta files like this:
```
-rw-r--r-- 1 heikki heikki  25575424 May 17 20:35 000000000000000000000000000000000000-FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF__0000000001696070-00000000016960E9
-rw-r--r-- 1 heikki heikki   8691712 May 17 20:35 000000000000000000000000000000000000-FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF__00000000016960E9-00000000016E4DB9
-rw-r--r-- 1 heikki heikki 486014976 May 17 20:35 000000000000000000000000000000000000-FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF__00000000016E4DB9-000000000BFCE3E1
-rw-r--r-- 1 heikki heikki     32768 May 17 20:35 000000000000000000000000000000000000-FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF__000000000BFCE3E1-000000000BFD0FE9
-rw-r--r-- 1 heikki heikki 273997824 May 17 20:35 000000000000000000000000000000000000-FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF__0000000060045901-000000007005EAC1
-rw-r--r-- 1 heikki heikki 298672128 May 17 20:35 000000000000000000000000000000000000-FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF__000000007005EAC1-0000000080062E99
-rw-r--r-- 1 heikki heikki 277823488 May 17 20:35 000000000000000000000000000000000000-FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF__0000000080062E99-000000009007F481
-rw-r--r-- 1 heikki heikki 272793600 May 17 20:35 000000000000000000000000000000000000-FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF__000000009007F481-00000000A009F7C9
-rw-r--r-- 1 heikki heikki 172220416 May 17 20:35 000000000000000000000000000000000000-FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF__00000000A009F7C9-00000000AA284EB9
-rw-r--r-- 1 heikki heikki     32768 May 17 20:35 000000000000000000000000000000000000-FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF__00000000AA286471-00000000AA2886B9
```
Note that gap between LSNs 000000000BFD0FE9 and 0000000060045901. Looking at the code in the `compact_level0` function, it will do bad things for a set of files like that. It will take all of those files, and create compacted Level 1 files from them. The new files will have LSN range 0000000001696070-00000000AA2886B9, i.e the min and max of the L0 files, But because of the gap, the new files will miss any records in the gap.

**Comment by hlinnaka** [comment]
Fix: https://github.com/neondatabase/neon/pull/1733

**Comment by hlinnaka** [comment]
Fixed by commit ee3bcf108d0ed1c1442c22182dcaaa1a6c518df4.
